### PR TITLE
Support authentication when calling feast

### PR DESCRIPTION
**Pull request by tiopramayudi**
**What this PR does / why we need it**:

Feast already support authentication, hence we need to update our feast core client to pass required credentials. Also we need to update `Authorization-Header` when calling feast from UI

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/cmd/api/setup.go [hunk 1]
```diff
@@ -83,8 +84,13 @@ func initMLPAPIClient(ctx context.Context, cfg config.MlpAPIConfig) mlp.APIClien
...
 
-func initFeastCoreClient(feastCoreURL string) core.CoreServiceClient {
-	cc, err := grpc.Dial(feastCoreURL, grpc.WithInsecure())
+func initFeastCoreClient(feastCoreURL, feastAuthAudience string) core.CoreServiceClient {
+	cred, err := feast.NewGoogleCredential(feastAuthAudience)
+	if err != nil {
+		panic(err)
+	}
+
+	cc, err := grpc.Dial(feastCoreURL, grpc.WithInsecure(), grpc.WithPerRPCCredentials(cred))
```
Comment (pradithya) [inline api/cmd/api/setup.go:93]: You have to choose either `grpc.WithInsecure()` or `grpc.WithPerRPCCredentials(cred)` isn't it?
And we should add enabled/disabled flag

File: api/cmd/transformer/main.go [hunk 2]
```diff
@@ -70,7 +70,14 @@ func main() {
 	if err != nil {
 		logger.Fatal("Unable to parse Feast Serving Port", zap.String("feast-serving-port", feastPort), zap.Error(err))
 	}
-	feastClient, err := feastSdk.NewGrpcClient(feastHost, feastPortInt)
+
+	cred, err := feastSdk.NewGoogleCredential(cfg.Feast.AuthAudience)
```
Comment (pradithya) [inline api/cmd/transformer/main.go:74]: How is it different from `initFeastCoreClient` ?

Comment (pradithya) [inline api/cmd/transformer/main.go:74]: We don't need auth for feast serving

File: charts/merlin/templates/merlin-deployment.yaml [hunk 3]
```diff
@@ -143,6 +143,10 @@ spec:
             value: "{{ .Values.merlin.transformer.feast.servingURL }}"
           - name: FEAST_CORE_URL
             value: "{{ .Values.merlin.transformer.feast.coreURL }}"
+          - name: FEAST_CORE_AUTH_AUDIENCE
+            value: "{{ .Values.merlin.transformer.feast.coreAuthAudience }}"
+          - name: FEAST_SERVING_AUTH_AUDIENCE
```
Comment (pradithya) [inline charts/merlin/templates/merlin-deployment.yaml:148]: We don't need auth for serving

File: api/cmd/api/setup.go (hunk 1 above)
Comment (tiopramayudi) [reply api/cmd/api/setup.go:93]: need to add `gcpc.WIthInsecure()` because the feast credentials is not transport credentials.

Comment (pradithya) [reply api/cmd/api/setup.go:93]: Got it

**Pull request merged**